### ISSUE TITLE
merge.data.table throws error for `<` in variable names

### ISSUE THREAD

**Issue by mplatzer** [Rdatatable/data.table#2140]
an error is thrown if one wants to merge a data.table with `<` or `>` in its variable names.

    > dt <- data.table('x <TBD>' = 1:2)
    > merge(dt, dt)
    Error in `[.data.table`(y, x, nomatch = if (all.x) NA else 0, on = by,  : Column(s) [x ] not found in x

using `merge(dt, dt, by = 'x <TBD>')` results in the same error. If angle brackets are not supported by data.table, I would suggest adapting the error message.

sidenote: this works with data.frames

    > df <- data.frame('x <TBD>' = 1:2)
    > merge(df, df)
      x..TBD.
    1       1
    2       2

**Comment by MichaelChirico** [comment]
Actually, you'll notice that the comparison with `data.frame` is not exactly correct:

```
DF = data.frame('x <TBD>' = 1:2)
# the name has already been fixed
DF
#   x..TBD.
# 1       1
# 2       2
# so if we try and merge with data.table on this it works fine
setDT(DF)
merge(DF, DF)
#    x..TBD.
# 1:       1
# 2:       2
```

That's not exactly an answer... still investigating. But basically, `merge.data.table` is interpreting this using the more flexible `on = .(x <= 5)` notation and thinking you mean `x < TBD`. It's a strange corner case, so not sure it's worth controlling for...

**Comment by MichaelChirico** [comment]
@arunsrinivasan is `on` supposed to accept more than one logical operator? `on = 'y <= foo'` is documented behavior, what about `on = 'foo <= y <= bar'`?

The two things that may be worth erroring for in this case are: (1) `<` and `>` are in opposition. Only `<`/`<=`/`==` or `>`/`>=`/`==` make sense with normal parsing of logical statements (2) There's nothing after the second logical operator (if we parsed this logically it could be `x`:`<`:`TBD`:`>`:` `).

**Comment by MichaelChirico** [comment]
Barring that, we could just give an error & suggestion to rename columns that have logical operators present.

**Comment by arunsrinivasan** [comment]
@MichaelChirico no, `foo <= y <= bar` isn't supported yet. But could be useful.
